manifests/manifest.FIREFOX.json

```diff
@@ -0,0 +1,17 @@
+{
+  "background": {
+    "scripts": [
+      "build/background.js"
+    ]
+  },
+  "browser_specific_settings": {
+    "gecko": {
+      "id": "{6bc19aa2-7d91-4791-b5d4-8d84b5f3cdad}",
+      "strict_min_version": "109.0"
+    }
+  },
+  "optional_permissions": [
```

Comment:
Now that I've had time to review the code, I'm (personally) much more comfortable giving full permissions to the extension. Given the annoyance of the green dot I'll revert the permission changes. 👍🏼

Reply:
Testing this in Firefox, it still seems to treat the `host_permissions` as optional, which means the connection test request fails, and the dot still shows up. The [docs](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/host_permissions) kind of elude to this:
> Most browsers treat host_permission as optional. If you request permissions using this key, users may get prompted to grant those permissions during installation. As of June 2023, Safari, Firefox, and some Chromium-based browsers don't prompt the user during installation.

What a weird way to go about it rather than just asking during installation. Also what is the point of optional host permissions?

Anyway, It seems you can request host permissions for the same wildcard URLs that are specified in the manifest, which then grants the extension to access all URLs:
```js
    // Request host permissions
    const granted = await browserAPI.permissions.request({
      origins: [`https://*/*`, `http://*/*`],
    });
```

However I'm not sure if that is intentional or a bug, also haven't checked how Chrome handles this.

At this point I'm not sure if I want to continue migrating FF to v3, there really isn't a benefit regarding permissions for users so far, and it would require this weird approach above with the delayed permission grant.

Reply:
Oh the joys of self-hosting 😄 

Alright I'm going to put this PR back into draft and start on PRs for the other features I've worked on. We'll see how far we can get before this migration (or lack thereof) limits us.